src/main/kotlin/org/jetbrains/research/testspark/helpers/psiHelpers/JavaPsiHelper.kt

```diff
@@ -173,145 +136,313 @@ class JavaPsiHelper : PsiHelperInterface {
 
-    /**
-     * Checks if a PSI method is a default constructor.
-     *
-     * @param psiMethod the PSI method of interest
-     * @return true if the PSI method is a default constructor, false otherwise
-     */
-    private fun isDefaultConstructor(psiMethod: PsiMethod): Boolean {
-        return psiMethod.isConstructor && psiMethod.body?.isEmpty ?: false
-    }
+    override val allMethods: List<PsiMethodWrapper> get() = psiClass.allMethods.map { JavaPsiMethodWrapper(it) }
 
-    /**
-     * Checks if a PSI class is an abstract class.
-     *
-     * @param psiClass the PSI class of interest
-     * @return true if the PSI class is an abstract class, false otherwise
-     */
-    private fun isAbstractClass(psiClass: PsiClass): Boolean {
-        if (psiClass.isInterface) return false
+    override val qualifiedName: String get() = psiClass.qualifiedName ?: ""
+
+    override val text: String get() = psiClass.text
 
-        val methods = PsiTreeUtil.findChildrenOfType(psiClass, PsiMethod::class.java)
-        for (psiMethod: PsiMethod in methods) {
-            if (!isMethodConcrete(psiMethod)) {
-                return true
+    override val fullText: String
+        get() {
+            var fullText = ""
+            val fileText = psiClass.containingFile.text
+
+            // get package
+            packagePattern.findAll(fileText, 0).map {
+                it.groupValues[0]
+            }.forEach {
+                fullText += "$it\n\n"
+            }
+
+            // get imports
+            importPattern.findAll(fileText, 0).map {
+                it.groupValues[0]
+            }.forEach {
+                fullText += "$it\n"
             }
+
+            // Add class code
+            fullText += psiClass.text
+
+            return fullText
         }
 
-        // check if a class is noted as abstract in the text
-        return psiClass.text.replace(" ", "")
-            .contains("abstractclass${psiClass.name}", ignoreCase = true)
+    override val superClass: PsiClassWrapper? get() = psiClass.superClass?.let { JavaPsiClassWrapper(it) }
+
+    override val virtualFile: VirtualFile get() = psiClass.containingFile.virtualFile
+
+    override val containingFile: PsiFile get() = psiClass.containingFile
+
+    override fun searchSubclasses(project: Project): Collection<PsiClassWrapper> {
+        val scope = GlobalSearchScope.projectScope(project)
+        val query = ClassInheritorsSearch.search(psiClass, scope, false)
+        return query.findAll().map { JavaPsiClassWrapper(it) }
     }
 
+    override fun getInterestingPsiClassesWithQualifiedNames(
+        psiMethod: PsiMethodWrapper,
+    ): MutableSet<PsiClassWrapper> {
+        val interestingMethods = mutableSetOf(psiMethod as JavaPsiMethodWrapper)
+        for (currentPsiMethod in allMethods) {
+            if ((currentPsiMethod as JavaPsiMethodWrapper).isConstructor) interestingMethods.add(currentPsiMethod)
```

Comment:
It looks like it will work the same way if `as JavaPsiMethodWrapper` is removed. Correct me if I am wrong.

Reply:
It will not work because then it will be treated as PsiMethodWrapper and since in this Interface we don't have a method parameterList, we will have a problem in
`methodIt.parameterList.parameters.forEach { paramIt ->`